backend/main.py

```
@@ -1265,8 +1271,53 @@ async def get_task_context(project_id: str, session_id: str):
+        return {
+            "success": True,
+            "message": "Codebase connected successfully",
+            "project_id": request.project_id,
+            "codebase_path": request.path
+        }
+        
+    except HTTPException:
+        raise
+    except Exception as e:
+        logger.error(f"Error connecting codebase: {e}")
+        raise HTTPException(status_code=500, detail=f"Failed to connect codebase: {str(e)}")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)